demo/csrc/c/benchmark_jpeg_decode.cpp

```diff
@@ -0,0 +1,85 @@
+#include <opencv2/core/types_c.h>
```

Comment:
Examples in the demo folder aim at showing users how to use the API. It's better to make an example as simple as possible.
So, I suggest opening another cpp file, 'jpeg_decoder.cpp', presenting how to use the APIs declared in `image_decoder.h`